From 8506bea941d00e5413e2ad8a96bcd62b0e9aefa3 Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <masahiroy@kernel.org>
Date: Sat, 22 Jul 2023 13:48:04 +0900
Subject: [PATCH 22/62] kbuild: rpm-pkg: build the kernel in-place for rpm-pkg

Currently, 'make rpm-pkg' always builds the kernel from the pristine
source tree in the ~/rpmbuild/BUILD/ directory.

Build the kernel incrementally just like 'make binrpm-pkg'.

Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---
 scripts/Makefile.package | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/Makefile.package b/scripts/Makefile.package
index be9602fa9..bf2da97f2 100644
--- a/scripts/Makefile.package
+++ b/scripts/Makefile.package
@@ -72,6 +72,7 @@ PHONY += rpm-pkg
 rpm-pkg: srpm = $(shell rpmspec --srpm --query --queryformat='%{name}-%{VERSION}-%{RELEASE}.src.rpm' kernel.spec)
 rpm-pkg: srcrpm-pkg
 	+rpmbuild $(RPMOPTS) --target $(UTS_MACHINE)-linux -rb $(srpm) \
+	--build-in-place --noprep \
 	--define='_smp_mflags %{nil}' --define='make $(MAKE)'
 
 # srcrpm-pkg
-- 
2.42.0

